### PR TITLE
Adding flexibility to integration tests to pull the base images from OCIR or OCR and more

### PR DESCRIPTION
**Pull request by vanajamukkara**
Integration tests can now pull the base images from OCIR or OCR, default is OCIR. 
Removing the hard coded base image names. 
base image(WLS, FWMINFRA, DB) names can now be set using env variables before running the tests, default values are image names on OCIR.
Some refactoring to remove duplicate code from test classes.

Jenkins results 
https://build.weblogick8s.org:8443/job/wko-tmp/48/
https://build.weblogick8s.org:8443/job/wko-tmp/49/
https://build.weblogick8s.org:8443/job/wko-tmp/47/
https://build.weblogick8s.org:8443/job/wko-tmp/46/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java [hunk 1]
```diff
@@ -58,6 +75,63 @@
...
+  public static final String DB_IMAGE_NAME
+      = BASE_IMAGES_REPO + "/" + Optional.ofNullable(System.getenv("DB_IMAGE_NAME"))
+      .orElse(BASE_IMAGES_REPO.equals(OCR_REGISTRY) ? OCR_DB_IMAGE_NAME : OCIR_DB_IMAGE_NAME);
+  public static final String DB_IMAGE_TAG = Optional.ofNullable(System.getenv("DB_IMAGE_TAG"))
+      .orElse(BASE_IMAGES_REPO.equals(OCR_REGISTRY) ? OCR_DB_IMAGE_TAG : OCIR_DB_IMAGE_TAG);
+
+  // For kind, replace repo name in image name with KIND_REPO, otherwise use the actual image name
+  // For example, image container-registry.oracle.com/middleware/weblogic:12.2.1.4 will be pushed/used as
+  // localhost:5000/middleware/weblogic:12.2.1.4 in kind and in non-kind cluster it will be used as is.
+  public static final String WEBLOGIC_IMAGE_TO_USE_IN_SPEC = KIND_REPO != null ? KIND_REPO
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java:125]: Can you change this to WEBLOGIC_IMAGE_LOCAL as we discussed yesterday.

**Review by anpanigr — APPROVED** [review]
lgtm

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java (hunk 1 above)
Comment (vanajamukkara) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java:125]: sure, I will change in next PR as I have to do the testing again.

**Review by sankarpn — APPROVED** [review]

**Comment by tbarnes-us** [comment]
I have just one general comment.  It looks to me like OCIR is special-cased/hard-coded throughout - at least based on the sprinkling of the term "ocir" and "OCIR" throughout the tests.  Can the tests actually work with _any_ repo, not specifically just OCIR?  If so, can the design be altered throughout to reflect this?

**Comment by vanajamukkara** [comment]
Yes, the tests can work with any repo, its just the name that could be misleading. Will use better env var in my next PR.

**Comment by tbarnes-us** [comment]
It's not just the env var names, it's also the Java method and var names.  Why not address them in this PR?

**Comment by vanajamukkara** [comment]
> It's not just the env var names, it's also the Java method and var names. Why not address them in this PR?

this PR is already merged. :-)

**Pull request merged**